src/app/pages/RuntimeTransactionsPage/index.tsx

```diff
@@ -78,6 +83,12 @@ export const RuntimeTransactionsPage: FC = () => {
 
   const transactions = data?.data.transactions
 
+  useEffect(() => {
+    if (!beforeDate || offset === 0) {
+      setBeforeDate(transactions?.[0].timestamp)
+    }
+  }, [beforeDate, offset, transactions])
```

Comment:
This doesn't work correctly:
- open https://06966d30.oasis-explorer.pages.dev/mainnet/sapphire/tx?page=100
- first loads page 100 (33min ago)
- useEffect triggers
- uses timestamp of 990th transaction and offsets it by another 100 pages (1h ago)

![Recording 2024-09-12 at 22 48 30](https://github.com/user-attachments/assets/d213ce8f-54af-4b26-89b5-3265b9c499a6)

Reply:
what a failure. Anyway there are much more edge cases when it can re-render. Still testing new version so far looks stable (updated PR desc accordingly)